itest/multi_send_test.go

```diff
@@ -0,0 +1,289 @@
+			},
+			{
+				Asset: assetPReq,
+			},
+		},
+	)
+
+	groupKeyX := firstBatch[0].AssetGroup.TweakedGroupKey
+	assetXTranche3Req.GroupKey = groupKeyX
+
+	groupKeyY := firstBatch[2].AssetGroup.TweakedGroupKey
+	assetYTranche3Req.GroupKey = groupKeyY
+
+	// In our second batch we create the third tranche of the grouped asset
+	// X and Y as well as a passive asset Q.
+	secondBatch := MintAssetsConfirmBatch(
+		t.t, t.lndHarness.Miner.Client, t.tapd,
+		[]*mintrpc.MintAssetRequest{
+			{
+				Asset: assetXTranche3Req,
+			},
+			{
+				Asset: assetYTranche3Req,
+			},
+			{
+				Asset: assetQReq,
+			},
+		},
+	)
+
+	// Now that we have the asset created, we'll make a new node that'll
+	// serve as the node which'll receive the assets.
+	secondTapd := setupTapdHarness(
+		t.t, t, t.lndHarness.Bob, t.universeServer,
+	)
+	defer func() {
+		require.NoError(t.t, secondTapd.stop(!*noDelete))
+	}()
+
+	var (
+		aliceTapd = t.tapd
+		bobTapd   = secondTapd
+	)
+
+	// We now want to send all units of X to Bob. Since we can't yet select
+	// coins by just group key, we just ask for all units of tranche 1 and
+	// then promote some of the passive assets to active assets.
+	scriptKey1, anchorInternalKeyDesc1 := DeriveKeys(t.t, bobTapd)
+	var (
+		assetXTranche1   = firstBatch[0]
+		assetXTranche2   = firstBatch[1]
+		assetP           = firstBatch[4]
+		assetXTranche3   = secondBatch[0]
+		assetQ           = secondBatch[2]
+		assetXTranche1ID asset.ID
+		assetXTranche3ID asset.ID
+	)
+	copy(assetXTranche1ID[:], assetXTranche1.AssetGenesis.AssetId)
+	copy(assetXTranche3ID[:], assetXTranche3.AssetGenesis.AssetId)
+
+	vPktFirstBatch := tappsbt.ForInteractiveSend(
+		assetXTranche1ID, assetXTranche1.Amount, scriptKey1, 0,
+		anchorInternalKeyDesc1, asset.V0, chainParams,
+	)
+	fundRespFirstBatch := fundPacket(t, aliceTapd, vPktFirstBatch)
+
+	// Now we collect all the active assets, which are all tranches of
+	// asset X.
+	activePackets := make([]*tappsbt.VPacket, 0, 3)
+	tranche1, err := tappsbt.Decode(fundRespFirstBatch.FundedPsbt)
+	require.NoError(t.t, err)
+	activePackets = append(activePackets, tranche1)
+
+	for _, packetBytes := range fundRespFirstBatch.PassiveAssetPsbts {
+		passivePacket, err := tappsbt.Decode(packetBytes)
+		require.NoError(t.t, err)
+
+		passiveAssetID, err := passivePacket.AssetID()
+		require.NoError(t.t, err)
+
+		if bytes.Equal(
+			passiveAssetID[:], assetXTranche2.AssetGenesis.AssetId,
+		) {
+			activePackets = append(activePackets, passivePacket)
+		}
+	}
+
+	vPktSecondBatch := tappsbt.ForInteractiveSend(
+		assetXTranche3ID, assetXTranche3.Amount, scriptKey1, 0,
+		anchorInternalKeyDesc1, asset.V0, chainParams,
+	)
+	fundRespSecondBatch := fundPacket(t, aliceTapd, vPktSecondBatch)
+	tranche3, err := tappsbt.Decode(fundRespSecondBatch.FundedPsbt)
+	require.NoError(t.t, err)
+
+	activePackets = append(activePackets, tranche3)
+
+	// We now need to update the outputs of the passive packets to also go
```

Comment:
aren't passive assets supposed to stay "passive" i.e not change?
if the passive assets are going to a different script key, aren't they effectively "active"?

maybe I just can't grasp the phrasing

Reply:
You're correct, if they go to a different script key, they become active. So in this test we're promoting some of the passive assets to active, which should in the end be reflected in the transfer output table.
Added some comments to make it more clear.